src/scripting_api/util.js

```diff
@@ -20,11 +20,753 @@ class Util extends PDFObject {
+      },
+      M: data => {
+        return (data.month + 1).toString();
+      },
+      EEEE: data => {
+        return data.cDate.toLocaleDateString(undefined, { weekday: "long" });
+      },
+      EEE: data => {
+        return data.cDate.toLocaleDateString(undefined, { weekday: "short" });
+      },
+      E: data => {
+        return (data.dayOfWeek + 1).toString();
+      },
+      e: data => {
+        return ((data.dayOfWeek + 6) % 7).toString();
+      },
+      DD: data => {
+        return data.day.toString().padStart(2, "0");
+      },
+      D: data => {
+        return data.day.toString();
+      },
+      JJJ: data => {
+        const day = Math.floor(
+          (data.cDate - new Date(data.year, 0, 0)) / 86400000
+        );
+        return day.toString().padStart(3, "0");
+      },
+      J: data => {
+        return Math.floor((data.cDate - new Date(data.year, 0, 0)) / 86400000);
+      },
+      YYYY: data => {
+        return data.year.toString();
+      },
+      YY: data => {
+        return (data.year % 100).toString().padStart(2, "0");
+      },
+      G: data => {
+        return data.cDate.toLocaleDateString(undefined, { era: "short" });
+      },
+      w: data => {
+        const date = new Date(data.cDate.getTime());
+        // Week start on monday.
+        // Date for the closest wednesday
+        date.setDate(data.day + 2 - ((data.dayOfWeek + 6) % 7));
+
+        // Week 1 of a month is the week containing the 3rd
+        // so get the wednesday in week 1.
+        const first = new Date(data.year, data.month, 3);
+        first.setDate(first.getDate() + 2 - ((first.getDay() + 6) % 7));
+
+        return 1 + Math.floor((date - first) / 604800000);
+      },
+      WW: data => {
+        const date = new Date(data.cDate.getTime());
+        // Weeks start on monday.
+        // Date for the closest thursday
+        date.setDate(data.day + 3 - ((data.dayOfWeek + 6) % 7));
+
+        // Week 1 of a year is the week containing January 4
+        // so get the thursday in week 1.
+        const first = new Date(data.year, 0, 4);
+        first.setDate(first.getDate() + 3 - ((first.getDay() + 6) % 7));
+
+        return 1 + Math.floor((date - first) / 604800000);
+      },
+      HH: data => {
+        return data.hours.toString().padStart(2, "0");
+      },
+      H: data => {
+        return data.hours.toString();
+      },
+      hh: data => {
+        return (1 + ((data.hours + 11) % 12)).toString().padStart(2, "0");
+      },
+      h: data => {
+        return (1 + ((data.hours + 11) % 12)).toString();
+      },
+      kk: data => {
+        return (data.hours % 12).toString().padStart(2, "0");
+      },
+      k: data => {
+        return (data.hours % 12).toString();
+      },
+      KK: data => {
+        return (1 + ((data.hours + 23) % 24)).toString().padStart(2, "0");
+      },
+      K: data => {
+        return (1 + ((data.hours + 23) % 24)).toString();
+      },
+      mm: data => {
```

Comment:
It's conflicting with he `MM` for month... I don't really know how to handle that.

Reply:
I tried to check how acrobat (last version) is dealing with that stuff and astonishingly it doesn't handle the 3rd argument (i.e. bXFAPicture), so it's quite confusing.

Reply:
If Acrobat isn't handling it at all and there is confusion in the specification, I'd really like to remove it completely from this patch. It will be more consistent with Acrobat (our reference implementation) and that avoids us from filling in any blanks we can't verify. Moreover, I don't think this functionality is actually used often, and it would make the overall patch considerable smaller. It can always be added again in a follow-up PR.